docs/developer-experience/readme.md

```diff
@@ -0,0 +1,231 @@
+- Since this is not a tutorial, the explanations in this section differ a bit from the common pattern "piece of code" --> "what stuff does each line". 
+
+## What is the purpose of bauta.js?
+
+Bauta.js main goal is to provide a layer of utilities and abstractions between a low level library server and your server code. 
+
+To do his job, bauta.js can be used on top of two low level frameworks: express or fastify. Thus, bauta.js itself is an utility library that provides a set of useful abstractions and glues common patterns that cover most of the use cases related to the creation of REST API's and middlewares. 
+
+## Ideal Scenario
+
+The ideal scenario where bauta.js shines is in delivering REST API's and especially using micro-services architecture. This allows you to use and reuse the bauta.js abstractions in different services and gain back the time and effort that you might have invested in learning to use it. 
+
+Thus, the ideal scenario could consist:
+
+- A set of one or more services, every one of them with its own business needs. Following the micro-services architecture you deploy each of those services independently.
+- Those same set of services should share a common set of technical requirements in terms of security, token management, external calls management, error management, logging, etc.
+
+### What about other scenarios?
+
+Depending on the scenario is up to you to decide whether bauta.js is useful to you or not. We hope that this document can provide information to easily assist you in that decision.
+
+For example, if you are exposing a GraphQL server then you can decide to use directly a GraphQL server like Apollo or Mercurius and you may think that you do not need bauta.js [^1].
+
+Another case would be if your REST API is an entire monolith. Is it worth it to learn bauta.js if you are maintaining an already big monolith? As usual, it depends, but the advantage of having your monolith in bauta.js is that it should be easier to separate it in different micro-services if you have the need to do so in the future. 
+
+## The key concepts of bauta.js
+
+### StepFunction: the working unit of bauta.js
+
+StepFunction is a function in bauta.js that conforms to the following signature:
+
+- it has three input parameters: prev, context and bauta.js instance itself.
+- it returns a value or a promise that would eventuall return a value.
+
+#### Why it is so important?
+
+Because not only the application code uses this as a type. Bauta.js itself uses this signature for its internal execution. Even bauta.js more advanced abstractions are in turn or return a StepFunction. And finally, because almost all the logic defined in terms of endpoints are a combination of one or more StepFunctions (check pipeline for more details).
+
+One could Think about the code of a bauta.js application as a bunch of StepFunctions that are run one after another or in parallel and whose results are passed from one to the next defined in your code. 
```

Comment:
```suggestion
Because not only the application code uses this as a type. Bauta.js itself uses this signature for its internal execution. Even bauta.js more advanced abstractions are, in turn, or return, a `StepFunction`. And finally, almost all the logic defined in terms of endpoints is a combination of one or more step functions (check pipeline for more details).

One could Think about the code of a bauta.js application as a bunch of step functions run one after another or in parallel and whose results are passed from one to the next defined in your code. 
```